_posts/2024-08-02-gsoc2024_talagayev.md

```diff
@@ -0,0 +1,193 @@
+indices of the molecules, but as more and more features were added the visualization grew bigger and had more features that it needed to display.
+
+### Feature Addition
+
+The feature addition was the next step that I was working on and was mainly the biggest part of the project since there was always an additional interesting feature that could be added
+that was proposed either by myself or one of the mentors.
+
+The first feature addition that I was working on were the physiochemical properties as well as the number of hydrogen bond donors and acceptors of the molecule.
+
+After it was done I started adding features, which would by selecting the checkbox highlight the features mentioned in the checkbox. For example, I implemented pharmacophore checkboxes,
+which by selection would highlight the selected pharmacophores, thus by selecting a hydrophobic feature, the atoms of the molecule that are hydrophobic via the `RDKit` pharmacophore
+recognition were highlighted in the assigned colors.
+
+Here is a short summary of features that I added for the visualization, which would show a value for the corresponding feature or adjust the figure of the molecule to display the values:
+
+- **Phyisochemical Properties**, **Atom Indices**, **Partial Charges**, **Hydrogen Bond Donors&Acceptors**
+
+And here is a short summary of features, that will highlight a specific atoms and bonds corresponding to their features if they are selected:
+
+- **Rotatable Bonds**, **Partial Charge Heatmap**, **Functional Groups**, **Stereocenters**, **Murcko Scaffold**, **Pharmacophores**
+
+Here I would also address that for the pharmacophores there is a separate checkbox for each of the possible pharmacophore features, thus you can select the specific pharmaocphore feature
+that needs to be highlighted.
+
+For the functional groups it currently uses SMART Patterns to identify certain functional groups and will then generate the checkboxes of the functional groups that were recognized and
+by selecting them you are able to see the atoms of this functional group.
+
+### Code Restructure
+
+This is also a section that was done throughout the whole project since in the beginning, the code was mainly consisting of functions in one file and this was one of the points that I was
+glad that my mentors helped me with since the code restructure allowed me to go from a code that was first based upon only function to structure it with classes and then in the end the
+separate classes were moved into separate files, thus a clear structure of the code was created, where every file was distinctive from the other and had an important purpose and a clear
+structure 
+
+### Actions, Testing and Documentation
+
+This would be the final part of what I did and here I would put all these three parts together. Pytests were created for this project in addition to the code being uploaded to
```

Comment:
Pytests is the name of the framework used for testing, I wouldn't use it as a noun, you could instead say:
```suggestion
This would be the final part of what I did and here I would put all these three parts together. A testsuite based on the [pytest](https://docs.pytest.org/) framework was created for this project, in addition to the code being uploaded to
```